client/app/components/queries/schedule-dialog.js

```diff
@@ -12,94 +26,229 @@ function padWithZeros(size, v) {
+  DROPDOWNS.forEach((dropdown) => {
+    templateString += `<select ng-model=${dropdown.type} ng-change="updateSchedule()" ng-options="c as c for c in ${dropdown.options}"></select>`;
+  });
+  templateString += '</label>';
+  return templateString;
+}
+
+function setTimePickerIntervalValues($scope) {
+  $scope.hourOptions = map(range(0, 24), partial(padWithZeros, 2));
+  $scope.minuteOptions = map(range(0, 60, 5), partial(padWithZeros, 2));
+
+  if ($scope.query.hasScheduleTime()) {
+    const parts = $scope.query.scheduleInLocalTime().split(':');
+    $scope.minute = parts[1];
+    $scope.hour = parts[0];
+  }
+}
+
+function queryTimePicker() {
+  return {
+    restrict: 'E',
+    scope: {
+      query: '=',
+    },
+    template: makeTimePickerTemplateString(),
+    link($scope) {
+      setTimePickerIntervalValues($scope);
 
-      $scope.$watch('refreshType', () => {
-        if ($scope.refreshType === 'daily') {
-          $scope.updateSchedule();
+      $scope.$watch('$parent.showTimePicker', () => {
+        if ($scope.$parent.showTimePicker && (!$scope.minute || !$scope.hour)) {
+          $scope.minute = '15';
+          $scope.hour = '00';
+        } else if (!$scope.$parent.showTimePicker) {
+          $scope.minute = null;
+          $scope.hour = null;
         }
+        $scope.updateSchedule();
+        return $scope.$parent.showTimePicker;
       });
+
+      $scope.updateSchedule = () => {
+        let newSchedule = null;
+        if ($scope.hour && $scope.minute) {
+          newSchedule = moment().hour($scope.hour)
+            .minute($scope.minute)
+            .utc()
+            .format('HH:mm');
+        }
+        if (newSchedule !== $scope.query.schedule_time) {
+          $scope.query.schedule_time = newSchedule;
+        }
+      };
     },
   };
 }
 
-function queryRefreshSelect(clientConfig) {
+function makeDayPickerTemplateString() {
+  let templateString = `
+    <div class="btn-toolbar"><div class="btn-group" data-toggle="buttons">`;
+  WEEKDAYS_SHORT.forEach((day) => {
+    templateString += `<button class="btn btn-xs btn-default" ng-class="{active: days == '${day}'}" ng-click="updateSchedule('${day}')" value="${day}">${day}</button>`;
+  });
+  templateString += '</div></div></label>';
+  return templateString;
+}
+
+function queryDayPicker() {
   return {
     restrict: 'E',
     scope: {
-      refreshType: '=',
       query: '=',
-      saveQuery: '=',
     },
-    template: `<select
-                ng-disabled="refreshType != 'periodic'"
-                ng-model="query.schedule"
-                ng-change="saveQuery()"
-                ng-options="c.value as c.name for c in refreshOptions">
-                <option value="">No Refresh</option>
-                </select>`,
+    template: makeDayPickerTemplateString(),
     link($scope) {
-      $scope.refreshOptions =
-        clientConfig.queryRefreshIntervals.map(interval => ({ value: String(interval), name: `Every ${durationHumanize(interval)}` }));
```

Comment:
I think it would be possible and not *too* complicated to only show allowable intervals specified by the admin. The only concern is that the way it works now, the admin would have a *much* longer list to show that 1, 2, 3....X hours/days/weeks are acceptable here: https://github.com/getredash/redash/blob/master/redash/settings/__init__.py#L205

I think it would be easier to use a set of allowable frequencies as you suggested.